gnpy/core/equipment.py

```diff
@@ -364,6 +368,23 @@ def update_dual_stage(equipment):
                 exit()
     return equipment
 
+def roadm_restrictions_sanity_check(equipment):
+    # verifies that booster and preamp restrictions specified in roadm equipment are listed
+    # in the edfa.
+    restrictions = equipment['Roadm']['default'].restrictions['booster_variety_list'] + \
+        equipment['Roadm']['default'].restrictions['preamp_variety_list']
+    for amp_name in restrictions:
+        try:
+            if equipment['Edfa'][amp_name]:
+                pass
+        except KeyError:
+            print(
+                f'\x1b[1;31;40m'\
```

Comment:
I understand your concern, except that it's more an "EquipmentError" than a "TopologyError" since it is just checking that the restriction in equipment_config json is valid.
I prepare this change